cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -120,26 +122,18 @@ func (p *BicepProvider) Initialize(ctx context.Context, projectPath string, opti
 	return err
 }
 
+var ErrEnsureEnvPreReqBicepCompileFailed = errors.New("")
+
 // EnsureEnv ensures that the environment is in a provision-ready state with required values set, prompting the user if
-// values are unset.
-//
-// An environment is considered to be in a provision-ready state if it contains both an AZURE_SUBSCRIPTION_ID and
-// AZURE_LOCATION value. Additionally, for resource group scoped deployments, an AZURE_RESOURCE_GROUP value is required.
+// values are unset. This also requires that the Bicep module can be compiled.
 func (p *BicepProvider) EnsureEnv(ctx context.Context) error {
```

Comment:
Why not capture subscription and location before attempting to compile?

Reply:
The location depends on the `Location` filter metadata in the `bicep` file, this was a feature @vhvb1989  added awhile back. So we need to compile first.